static/js/components/User.js

```diff
@@ -5,18 +5,19 @@ import IconButton from 'react-mdl/lib/IconButton';
 import iso3166 from 'iso-3166-2';
 
 import { makeProfileImageUrl } from '../util/util';
+import EmploymentForm from './EmploymentForm';
 
 export default class User extends React.Component {
   static propTypes = {
-    profile: React.PropTypes.object.isRequired
+    profile: React.PropTypes.object
   };
 
   render() {
     const { profile } = this.props;
 
-    let getStateName = () => {
-      return iso3166.subdivision(profile.state_or_territory).name;
-    };
+    let getStateName = () => (
+      iso3166.subdivision(profile.state_or_territory).name
```

Comment:
I ran into an error here when I went to a missing profile page. I think we need some code in this page to at least guard if `profile.state_or_territory` is undefined. I'm not sure how we want our 404 page to look like or function (cc @pdpinch)